core/src/main/scala/za/co/absa/spline/harvester/builder/write/WriteCommandExtractor.scala

```diff
@@ -78,17 +78,24 @@ class WriteCommandExtractor(pathQualifier: PathQualifier, session: SparkSession)
         }
 
       case cmd: InsertIntoHadoopFsRelationCommand =>
-        val path = cmd.outputPath.toString
-        val qPath = pathQualifier.qualify(path)
-        val fileFormat = cmd.fileFormat
-        fileFormat match {
-          case SparkAvroSourceExtractor(avro) =>
-            WriteCommand(cmd.nodeName, SourceIdentifier(Some("Avro"), qPath), cmd.mode, cmd.query, cmd.options)
-          case DatabricksAvroSourceExtractor(avro) =>
-            WriteCommand(cmd.nodeName, SourceIdentifier(Some("Avro"), qPath), cmd.mode, cmd.query, cmd.options)
-          case _ =>
-            val format = fileFormat.toString
-            WriteCommand(cmd.nodeName, SourceIdentifier(Some(format), qPath), cmd.mode, cmd.query, cmd.options)
+        val catalogTable = cmd.catalogTable.orNull
+        if(catalogTable != null){
+          val mode = if (cmd.mode == SaveMode.Overwrite) Overwrite else Append
+          asTableWriteCommand(cmd.nodeName, catalogTable, mode, cmd.query)
+        }
```

Comment:
Minor suggestion:
Since `catalogTable` is an `Option` monad, there is no need to deal with `null`s.
```scala
        cmd.catalogTable
          .map(catalogTable => {
            val mode = if (cmd.mode == SaveMode.Overwrite) Overwrite else Append
            asTableWriteCommand(cmd.nodeName, catalogTable, mode, cmd.query)
          })
          .getOrElse({
            ...
          })
```

Reply:
Thanks for this suggestion. I'll try to implement this format instead of the if conditions.

Will also work on getting more encompassing test cases together. There is already an original one started called InsertIntoTest that I will expand more.